### ISSUE TITLE
Local PV test is flaky

### ISSUE THREAD

**Issue by msau42** [kubernetes/kubernetes#51995]
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug
/sig storage
/priority important-soon
cc @kubernetes/sig-storage-bugs 

**What happened**:
Local PV test has flaked twice today due to PVC failing to bind.

[log](https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-alpha-features/15441)
[testgrid](https://k8s-testgrid.appspot.com/sig-storage#gci-gce-alpha)

Looking at the [log](https://storage.googleapis.com/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-alpha-features/15441/artifacts/bootstrap-e2e-master/kube-controller-manager.log), it seems like the bind failed because the PV object was out of date, but we never resynced the PV object.

```
I0905 22:41:16.205501      19 pv_controller.go:780] updating PersistentVolume[local-pvxhlvt]: binding to "e2e-tests-persistent-local-volumes-test-9t4zw/pvc-g5swm" failed: Operation cannot be fulfilled on persistentvolumes "local-pvxhlvt": the object has been modified; please apply your changes to the latest version and try again
I0905 22:41:16.205545      19 pv_controller.go:864] error binding volume "local-pvxhlvt" to claim "e2e-tests-persistent-local-volumes-test-9t4zw/pvc-g5swm": failed saving the volume: Operation cannot be fulfilled on persistentvolumes "local-pvxhlvt": the object has been modified; please apply your changes to the latest version and try again
I0905 22:41:16.205561      19 pv_controller_base.go:233] could not sync claim "e2e-tests-persistent-local-volumes-test-9t4zw/pvc-g5swm": Operation cannot be fulfilled on persistentvolumes "local-pvxhlvt": the object has been modified; please apply your changes to the latest version and try again
```

**Comment by msau42** [comment]
cc @jsafrane

**Comment by msau42** [comment]
The last resync was before processing this volume:
```
I0905 22:41:05.200749      19 pv_controller_base.go:394] resyncing PV controller
```

Afterwards, there are no more resync messages.

**Comment by jsafrane** [comment]
Indeed, PV controller cannot save the PV at 22:41:16.205561. It should be fixed in the next periodic sync (every 15 seconds), but the controller is restarted in 5 seconds:

```
I0905 22:41:21.471767      34 pv_controller_base.go:259] Starting persistent volume controller
```
This new controller never received the PV or PVC from informers. Where is this PV and PVC lost?


BTW, controller-manager crashes because of resize controller, I filled #52012

**Comment by jsafrane** [comment]
PV controller does not get the PVs / PVCs because expand controller crashes regularly after start when informers are synced.

**Comment by jsafrane** [comment]
It should be fixed by #52018
